src/pymor/models/iosys.py

```diff
@@ -1735,189 +1763,13 @@ def to_matrices(self):
-        ----------
-        typ
-            The type of the Gramians used (see :meth:`LTIModel._sv_U_V`).
-        mu
-            |Parameter values|.
-
-        Returns
-        -------
-        sv
-            One-dimensional |NumPy array| of singular values.
-        Uh
-            |NumPy array| of left singular vectors as rows.
-        Vh
-            |NumPy array| of right singular vectors as rows.
-        """
-        return self.to_lti()._sv_U_V(typ=typ, mu=mu)
-
-    def hsv(self, mu=None):
-        """Hankel singular values.
-
-        .. note::
-            Assumes the system is asymptotically stable.
-
-        Parameters
-        ----------
-        mu
-            |Parameter values|.
-
-        Returns
-        -------
-        sv
-            One-dimensional |NumPy array| of singular values.
-        """
-        return self._sv_U_V(mu=mu)[0]
-
-    def h2_norm(self, mu=None):
-        """Compute the H2-norm.
-
-        .. note::
-            Assumes the system is asymptotically stable.
-
-        Parameters
-        ----------
-        mu
-            |Parameter values|.
-
-        Returns
-        -------
-        norm
-            H_2-norm.
-        """
-        return self.to_lti().h2_norm(mu=mu)
-
-    def hinf_norm(self, mu=None, return_fpeak=False, ab13dd_equilibrate=False):
-        """Compute the H_infinity-norm.
-
-        .. note::
-            Assumes the system is asymptotically stable.
-
-        Parameters
-        ----------
-        mu
-            |Parameter values|.
-        return_fpeak
-            Should the frequency at which the maximum is achieved should be returned.
-        ab13dd_equilibrate
-            Should `slycot.ab13dd` use equilibration.
-
-        Returns
-        -------
-        norm
-            H_infinity-norm.
-        fpeak
-            Frequency at which the maximum is achieved (if `return_fpeak` is `True`).
-        """
-        return self.to_lti().hinf_norm(mu=mu,
-                                       return_fpeak=return_fpeak,
-                                       ab13dd_equilibrate=ab13dd_equilibrate)
-
-    def hankel_norm(self, mu=None):
-        """Compute the Hankel-norm.
-
-        .. note::
-            Assumes the system is asymptotically stable.
-
-        Parameters
-        ----------
-        mu
-            |Parameter values|.
-
-        Returns
-        -------
-        norm
-            Hankel-norm.
-        """
-        return self.hsv(mu=mu)[0]
+        return J, R, G, P, S, N, E, Q
 
     def __add__(self, other):
-        """Add a |PHLTIModel|, an |LTIModel|, or a |SecondOrderModel|."""
-        if isinstance(other, LTIModel):
-            return self.to_lti() + other
-
-        if isinstance(other, SecondOrderModel):
-            return self.to_lti() + other.to_lti()
-
         if not isinstance(other, PHLTIModel):
-            return NotImplemented
+            return super().__add__(other)
 
         assert self.S.source == other.S.source
```

Comment:
This assert should be enough (the three below should be automatically satisfied).

Reply:
There is still an issue with the `contract` method and `source_id` when I call the `to_berlin_form()`.

Reply:
And if $Q\neq I$ phirka takes really long to execute, maybe we can discuss this in the meeting a bit. Also there are two versions how to choose $W$:

- $W = QV$
- $W = QV(V^\top Q V)^{-1}$

 and I am not super sure which one we should implement.

Reply:
The first one yes, the question which choice of $W$ I think is still unresolved. In my other implementation I am using the second version and this works well, so maybe we should just use that here as well.

Reply:
I propose that we implement the first version $W=QV$. I think it is a little bit "nicer" numerically because there is no inverse involved. If one wants to have an identity $E=I$, one can still multiply from the left with $E^{-1}$ after the algorithm has run. I'll try to take a look.

Reply:
If we opt for $W = QV$, it holds for the ROM:
- $\hat{E} = V^TQ^TEV = W^TEV$,
- $\hat{J} = V^TQ^TJQV = W^TJW$,
- $\hat{R} = V^TQ^TRQV = W^TRW$,
- $\hat{Q} = I$,
- $\hat{G} = V^TQ^TG = W^TG$,
- $\hat{P} = V^TQ^TP = W^TP$,
- $\hat{S} = S$ and $\hat{N} = N$.

I am currently not sure if `PHIRKAReductor` is implementing this correctly. I will check again next week.